src/engine/ScreenShotTaker.cs

```diff
@@ -47,24 +65,73 @@ public string TakeAndSaveScreenShot()
         if (error != Error.Ok)
         {
             GD.PrintErr("Saving screenshot failed ", error);
-            return null;
+            return;
         }
 
         GD.Print("Saved screenshot: ", path);
-        return path;
     }
 
     /// <summary>
-    ///   Takes an image of the current viewport
+    ///   Takes and saves a screenshot
     /// </summary>
-    /// <returns>The image</returns>
-    public Image TakeScreenshot()
+    private void TakeScreenshot()
     {
-        var image = GetViewport().GetTexture().GetData();
+        if (isCurrentlyTakingScreenshot)
+        {
+            GD.Print("Already in the process of taking a screenshot.");
+            return;
+        }
 
-        // TODO: do we always need this?
-        image.FlipY();
+        isCurrentlyTakingScreenshot = true;
+        FileHelpers.MakeSureDirectoryExists(Constants.SCREENSHOT_FOLDER);
 
-        return image;
+        // If ScreenFilter is active, turn it off before taking a screenshot.
+        if (ColourblindScreenFilter.Instance.Visible)
+        {
+            step = Steps.Start;
+            ScreenFilterScreenshotStepper();
+            return;
+        }
+
+        SaveScreenshot(GetViewportTextureImage());
+        isCurrentlyTakingScreenshot = false;
+    }
+
+    /// <summary>
+    ///   Invokes itself to:
+    ///   1: Hide the ScreenFilter and wait a frame.
+    ///   2: Wait another frame.
+    ///   3: Take the screenshot, show the filter and wait a frame so the filter can show itself again faster.
```

Comment:
Is it actually needed to wait this frame?

Reply:
Yes, it's needed to wait 2 frames. Tested it, and Godot doc says the same.

Reply:
But I'm talking about waiting the one extra frame afterwards? I guess it doesn't matter if the extra frame *only* prevents another screenshot from happening. I was thinking that it would be bad if it prevented the game from working for one extra frame, but I guess that isn't the case after all.